src/material-experimental/mdc-button/button-base.ts

```diff
@@ -75,30 +75,23 @@ export const _MatButtonBaseMixin: CanDisableRippleCtor&CanDisableCtor&CanColorCt
 
 /** Base class for all buttons.  */
 export class MatButtonBase extends _MatButtonBaseMixin implements CanDisable, CanColor,
-                                                                  CanDisableRipple, OnChanges {
-  rippleTarget: RippleTarget = {
-    rippleConfig: {
-      animation: {
-        // TODO(mmalerba): Use the MDC constants once they are exported separately from the
-        // foundation. Grabbing them off the foundation prevents the foundation class from being
-        // tree-shaken. There is an open PR for this:
-        // https://github.com/material-components/material-components-web/pull/4593
-        enterDuration: 225 /* MDCRippleFoundation.numbers.DEACTIVATION_TIMEOUT_MS */,
-        exitDuration: 150 /* MDCRippleFoundation.numbers.FG_DEACTIVATION_MS */,
-      },
-    },
-    rippleDisabled: false,
+                                                                  CanDisableRipple {
+  rippleAnimation: RippleAnimationConfig = {
+    // TODO(mmalerba): Use the MDC constants once they are exported separately from the
+    // foundation. Grabbing them off the foundation prevents the foundation class from being
+    // tree-shaken. There is an open PR for this:
+    // https://github.com/material-components/material-components-web/pull/4593
+    enterDuration: 225 /* MDCRippleFoundation.numbers.DEACTIVATION_TIMEOUT_MS */,
+    exitDuration: 150 /* MDCRippleFoundation.numbers.FG_DEACTIVATION_MS */,
   };
 
-  /** The ripple renderer for the button. */
-  private _rippleRenderer =
-      new RippleRenderer(this.rippleTarget, this._ngZone, this._elementRef, this._platform);
+  /** Whether the ripple is centered on the button. */
+  isRippleCentered = false;
```

Comment:
can probably just hardcode this in the template `matRippleCentered=false`

Reply:
I think it's by intention to leave this a class member so that the derived classes (such as `MatIconButton)` can overwrite it.

Reply:
Right - one of the subclasses (icon button) is overriding this

Reply:
ah gotcha, I assume we don't want users messing with it though? so add leading `_`

Reply:
- [X] _